redfish-core/lib/power_subsystem.hpp

```diff
@@ -41,35 +82,26 @@ inline void getPowerSubsystemAllocation(
                 return;
             }
 
-            uint32_t powerCap{0};
-            bool powerCapEnable{false};
-            uint32_t maxPowerCapValue{0};
+            const uint32_t* powerCap = nullptr;
+            const bool* powerCapEnable = nullptr;
             const bool success = sdbusplus::unpackPropertiesNoThrow(
                 dbus_utils::UnpackErrorPrinter(), propertiesList, "PowerCap",
-                powerCap, "PowerCapEnable", powerCapEnable, "MaxPowerCapValue",
-                maxPowerCapValue);
+                powerCap, "PowerCapEnable", powerCapEnable);
 
             if (!success)
             {
                 messages::internalError(asyncResp->res);
                 return;
             }
 
-            // If MaxPowerCapValue valid, store Allocation properties in JSON
-            if ((maxPowerCapValue > 0) && (maxPowerCapValue < UINT32_MAX))
+            if ((powerCap != nullptr) && (powerCapEnable != nullptr))
             {
-                if (powerCapEnable)
+                if ((*powerCapEnable) && (*powerCap > 0) &&
+                    (*powerCap < UINT32_MAX))
```

Comment:
Since this is GET are the powerCap value checks necessary here? Should we just be returning what we got from D-Bus?

Since the variable itself is a uint32 it cannot be larger than UINT32_MAX can it?

Also the [D-Bus interface](https://github.com/openbmc/phosphor-dbus-interfaces/blob/bfcb5d0c39571cd5f0215d22e9f7fee0c23dfd59/yaml/xyz/openbmc_project/Control/Power/Cap.interface.yaml#L5) doesn't say anything about a minimum value.

Reply:
Good point on the variable checks.  Done.